src/range_proof.rs

```diff
@@ -24,12 +23,12 @@ pub fn range_proof<CS: ConstraintSystem>(
         // Enforce that a = 1 - b, so they both are 1 or 0.
         cs.constrain(a + (b - 1u64));
 
-        constraint.push((b, exp_2));
+        v = v - b * exp_2;
```

Comment:
it took me a while to realize that this operation actually appended `- b*exp_2` to the linear combination held by `v` (especially since the line below looks the same, but updates the holder value of `exp_2`). Maybe add a comment to make this clearer?

Reply:
Good point. We should also impl `AddAssign/SubAssign` to do `v += b*(-exp2)`.